int16 level
---
bool was_set
